### PR TITLE
Process UV rotation even in full transform

### LINKED ISSUE #3902: MGS Peace Walker: Indicator is inverted
pw1

Tested v0.8.1 and v0.9.1-1154 
Interpreter and JIT mode

### PR DESCRIPTION
**Pull request by unknownbrackets**
Assuming we don't expect these to be common, probably adding bits and etc. will just be a loss.  Or at least that's my excuse for just transforming inside the RotateUV func.

Fixes #3902.  I checked a few games which had broken before, and they work.  I don't think I have any games that use this feature, though.

-[Unknown]

### CODE REVIEW

File: GPU/Common/SoftwareTransformCommon.cpp [hunk 1]
```diff
@@ -58,18 +58,24 @@ static void SwapUVs(TransformedVertex &a, TransformedVertex &b) {
...
-	float y2 = v[0].y;
+static void RotateUV(TransformedVertex v[4], float flippedMatrix[16]) {
+	// Transform these two coordinates to figure out whether they're flipped or not.
+	Vec4f tl;
+	Vec3ByMatrix44(tl.AsArray(), v[2].pos, flippedMatrix);
+
+	Vec4f br;
+	Vec3ByMatrix44(br.AsArray(), v[0].pos, flippedMatrix);
+
+	float x1 = tl.x / tl.w;
```
Comment (hrydgard) [inline GPU/Common/SoftwareTransformCommon.cpp]: you could save two divisions by getting the reciprocals of tl.w and br.w and multiplying instead, although the perf improvement would probably be negligible..

Comment (unknownbrackets) [inline GPU/Common/SoftwareTransformCommon.cpp]: Okay, changed it.  Also added a not working software renderer change, but it wasn't working before either.  Hmm..

-[Unknown]

**Comment by lp200** [comment]
:smiley:

**Comment by thedax** [comment]
Always nice to see games getting fixes.

**Pull request merged**